### PR TITLE
Add Docker image for ROCm and documentation

### PR DESCRIPTION
**Pull request by kmaehashi**
Merge after uploading a wheel.

### CODE REVIEW

File: docs/source/install_rocm.rst [hunk 1]
```diff
@@ -17,20 +17,52 @@ The following ROCm libraries are required:
...
   $ sudo apt install hipblas hipsparse rocsparse rocrand rocthrust rocsolver rocfft hipcub rocprim rccl
 
+Environment Variables
+---------------------
+
+When running CuPy for ROCm, the following environment variables need to be set.
+
+* ``HCC_AMDGPU_TARGET``: ISA name supported by your GPU.
+  Run ``rocminfo`` and use the value displayed in ``Name:`` line (e.g., ``gfx900``).
+  You can specify comma-separated list of ISAs if you have multiple GPUs of different architecture.
```
Comment (leofang) [inline docs/source/install_rocm.rst:27]:
```suggestion
  You can specify a comma-separated list of ISAs if you have multiple GPUs of different architectures.
```

File: docs/source/install_rocm.rst [hunk 2]
```diff
@@ -17,20 +17,52 @@ The following ROCm libraries are required:
 
   $ sudo apt install hipblas hipsparse rocsparse rocrand rocthrust rocsolver rocfft hipcub rocprim rccl
 
+Environment Variables
+---------------------
+
+When running CuPy for ROCm, the following environment variables need to be set.
+
+* ``HCC_AMDGPU_TARGET``: ISA name supported by your GPU.
```
Comment (leofang) [inline docs/source/install_rocm.rst:25]: According to the ongoing discussion https://github.com/cupy/cupy/issues/4719#issuecomment-785981214 it seems we don't have to set `HCC_AMDGPU_TARGET` at runtime.

File: docs/source/install_rocm.rst [hunk 3]
```diff
@@ -17,20 +17,52 @@ The following ROCm libraries are required:
...
 ::
 
-  $ export HCC_AMDGPU_TARGET=gfx900  # This value should be changed based on your GPU
   $ export CUPY_INSTALL_USE_HIP=1
-  $ pip install cupy
-
-Note that ``HCC_AMDGPU_TARGET`` must be set to the ISA name supported by your GPU.
-Run ``rocminfo`` and use the value displayed in ``Name:`` line (e.g., ``gfx900``).
-
-You may also need to set ``ROCM_HOME`` (e.g., ``ROCM_HOME=/opt/rocm``).
```
Comment (leofang) [inline docs/source/install_rocm.rst:36]: I think we still need to set `ROCM_HOME` for the time being:
https://github.com/cupy/cupy/blob/f4751f1d4d145410e2268609c011fbf894775bc7/install/build.py#L45

**Review by leofang — CHANGES_REQUESTED** [review]

**Review by leofang — DISMISSED** [review]
LGTM except for one nitpick in language. I forgot that we still check `HCC_AMDGPU_TARGET` in `compiler.py` 😅 I can send a PR to remove it after the code freeze.

**Comment by kmaehashi** [comment]
Thanks @leofang!

**Comment by asi1024** [comment]
LGTM!

**Pull request merged**